common/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/common/BadRow.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2019 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.storage.bigquery.common
+
+import cats.data.NonEmptyList
+import com.snowplowanalytics.iglu.client.ClientError.ResolutionError
+import com.snowplowanalytics.iglu.core.{SchemaKey, SelfDescribingData}
+import com.snowplowanalytics.iglu.schemaddl.bigquery
+import com.snowplowanalytics.snowplow.analytics.scalasdk.Event
+import io.circe.syntax._
+import io.circe.{Encoder, Json}
+
+/**
+  * Class representing an enriched event that failed BigQuery Loader transformation,
+  * e.g. due unavailable schema, bug in Schema DDL or unexpected structure
+  * Unlike "failed insert" it did not get to BigQuery table
+  */
+sealed trait BadRow {
+  def getSelfDescribingData: SelfDescribingData[Json]
+  def compact: String  = getSelfDescribingData.asJson.noSpaces
+}
+
+object BadRow {
+  // TODO Enes: Get artifact name and version from project metadata
+  private val BQLoaderProcessorInfo = Processor("Snowplow BigQuery Loader", "0.2.0")
+
+  /**
+    * Represents the failure case where data can not be parsed as a proper event
+    * @param payload data blob tried to be parsed
+    * @param errors errors in the end of the parsing process
+    */
+  final case class EventParsingError(payload: String, errors: NonEmptyList[String]) extends BadRow {
+    def getSelfDescribingData: SelfDescribingData[Json] =
+      SelfDescribingData(BadRowSchemas.LoaderParsingError, (this: BadRow).asJson)
+  }
+
+  /**
+    * Represents the BigQuery specific failure cases
+    * @param enrichedEvent event which is enriched successfully
+    * @param errors info of failures
+    */
+  final case class BigQueryError(enrichedEvent: Event, errors: NonEmptyList[ErrorInfo]) extends BadRow {
+    def getSelfDescribingData: SelfDescribingData[Json] =
+      SelfDescribingData(BadRowSchemas.BigQueryError, (this: BadRow).asJson)
+  }
+
+  /**
+    * Gives info about the reasons of the BigQuery specific failures
+    */
+  sealed trait ErrorInfo
+
+  final case class UnexpectedValueInEvent(key: String) extends ErrorInfo
+
+  final case class IgluValidationError(error: ResolutionError) extends ErrorInfo
```

Comment:
This can be either resolution or validation error. `ClientError` represents both of them.